test/adaptive_load/plugin_util_test.cc

```diff
@@ -0,0 +1,378 @@
+  }
+  void UpdateAndRecompute(const nighthawk::adaptive_load::BenchmarkResult&) override {}
+  const double value_from_config_proto_;
+  const unsigned int value_from_command_line_options_template_;
+};
+
+/**
+ * A factory that creates a TestStepController from a LinearScoringFunctionConfig (see
+ * TestInputVariableSetter constructor).
+ */
+class TestStepControllerConfigFactory : public StepControllerConfigFactory {
+public:
+  std::string name() const override { return "nighthawk.test-step-controller"; }
+  Envoy::ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<nighthawk::adaptive_load::LinearScoringFunctionConfig>();
+  }
+  absl::Status ValidateConfig(const Envoy::Protobuf::Message& message) const override {
+    return DoValidateConfig(message);
+  }
+  StepControllerPtr createStepController(
+      const Envoy::Protobuf::Message& message,
+      const nighthawk::client::CommandLineOptions& command_line_options_template) override {
+    const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+    nighthawk::adaptive_load::LinearScoringFunctionConfig config;
+    Envoy::MessageUtil::unpackTo(any, config);
+    return std::make_unique<TestStepController>(config, command_line_options_template);
+  }
+};
+
+REGISTER_FACTORY(TestStepControllerConfigFactory, StepControllerConfigFactory);
+
+/**
+ * Creates an Any wrapping a TypedExtensionConfig for use in the |typed_config| of all test
+ * plugins in this file. The choice of the particular proto LinearScoringFunctionConfig is
+ * arbitrary. We don't leave the Any empty because we need to check that the plugin utils can
+ * correctly pass the proto through to the plugin.
+ */
+Envoy::ProtobufWkt::Any CreateTypedConfigAny(const double threshold) {
+  nighthawk::adaptive_load::LinearScoringFunctionConfig config;
+  config.set_threshold(threshold);
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  return config_any;
+}
+
+TEST(PluginUtilTest, CreatesCorrectInputVariableSetterType) {
+  envoy::config::core::v3::TypedExtensionConfig config;
+  config.set_name("nighthawk.test-input-variable-setter");
+  *config.mutable_typed_config() = CreateTypedConfigAny(0.0);
+  InputVariableSetterPtr plugin = LoadInputVariableSetterPlugin(config).value();
+  auto* typed_plugin = dynamic_cast<TestInputVariableSetter*>(plugin.get());
+  EXPECT_NE(typed_plugin, nullptr);
+}
+
+TEST(PluginUtilTest, ReturnsErrorFromInputVariableSetterConfigValidator) {
+  envoy::config::core::v3::TypedExtensionConfig config;
+  config.set_name("nighthawk.test-input-variable-setter");
+  *config.mutable_typed_config() = CreateTypedConfigAny(kBadConfigThreshold);
+  EXPECT_THAT(LoadInputVariableSetterPlugin(config).status().message(),
+              ::testing::HasSubstr("input validation failed"));
+}
+
+TEST(PluginUtilTest, PropagatesConfigProtoToInputVariableSetter) {
+  envoy::config::core::v3::TypedExtensionConfig config;
```

Comment:
(optional) Is this worth aliasing to make it shorter?

Reply:
Done